### PR TITLE
Added Safe (Gnosis) connector.

### PR DESCRIPTION
**Pull request by stas**
We've been having issues recently with WalletConnect so we worked on adding the official Safe connector to our dApp.

Wanted to share the implementation with you guys, hopefully this is useful.

Related convo #348 

---

### On testing the connector

The Safe connector is designed to get activated only when the dApp is loaded directly from the Safe.

Screenshot 2022-10-10 at 17-51-29 Safe

In order to enable the auto-connect support, a little trick is required (triggers the Safe connector if it's found):

```javascript
import React from 'react';
import { useConnect, createClient } from 'wagmi';
import { publicProvider } from 'wagmi/providers/public';

const { connect, connectors } = useConnect();
const wagmiClient = createClient({ connectors, publicProvider() });

// Auto-connect to Safe if available
React.useEffect(() => {
  const safeConnector = connectors.find((c) => c.id === 'safe' && c.ready);

  if (safeConnector) { 
    connect({ connector: safeConnector });
  } else {
    // Comment out to disable auto-connect...
    wagmiClient.autoConnect();
  }

}, [connect, connectors]);
```

### Custom Safe dApp integration

In order to make it easier to connect to Safe, you can add the dApp to their apps list and access it just as it would be a normal Safe app.

Screenshot 2022-10-10 at 17-50-39 Safe

You will need to add a `/manifest.json` file to your dApp. Here's an example:
```json
{
  "short_name": "Velodrome",
  "name": "Velodrome Finance",
  "description": "The Trading and Liquidity Marketplace on Optimism",
  "icons": [
    {
      "src": "logo.svg",
      "sizes": "any",
      "type": "image/svg+xml"
    }
  ],
  "start_url": ".",
  "display": "standalone",
  "theme_color": "#4862D8",
  "background_color": "#ffffff"
}
```

### CODE REVIEW

**Comment by stas** [comment]
Updated the integration snippet to solve the issue of `safe` auto-connect conflicting with default `autoConnect: true` client config.

**Comment by gmaijoe** [comment]
this would be great to get in! any chance we can get this merged @RainbowWalletAdmin @rainbow-infra ?

**Comment by outdoteth** [comment]
Any ETA on this?

**Comment by benjileibo** [comment]
eta on this?

File: packages/rainbowkit/src/wallets/getDefaultWallets.ts [hunk 1]
```diff
@@ -24,6 +25,7 @@ export const getDefaultWallets = ({
       wallets: [
         injectedWallet({ chains }),
         rainbowWallet({ chains }),
+        safeWallet({ chains }),
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/getDefaultWallets.ts:28]: Revert

File: packages/rainbowkit/src/wallets/getDefaultWallets.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import { coinbaseWallet } from './walletConnectors/coinbaseWallet/coinbaseWallet
 import { injectedWallet } from './walletConnectors/injectedWallet/injectedWallet';
 import { metaMaskWallet } from './walletConnectors/metaMaskWallet/metaMaskWallet';
 import { rainbowWallet } from './walletConnectors/rainbowWallet/rainbowWallet';
+import { safeWallet } from './walletConnectors/safeWallet/safeWallet';
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/getDefaultWallets.ts:9]: Revert

File: packages/rainbowkit/package.json [hunk 3]
```diff
@@ -61,8 +61,10 @@
     "vitest": "^0.5.0"
   },
   "dependencies": {
+    "@gnosis.pm/safe-apps-wagmi": "*",
     "@vanilla-extract/css": "1.9.1",
     "@vanilla-extract/dynamic": "2.0.2",
+    "@vanilla-extract/esbuild-plugin": "^2.2.0",
```
Comment (DanielSinclair) [inline packages/rainbowkit/package.json:67]: Revert

File: packages/rainbowkit/package.json [hunk 4]
```diff
@@ -51,8 +51,8 @@
     "@ethersproject/abstract-provider": "^5.5.1",
     "@ethersproject/providers": "^5.5.1",
     "@types/qrcode": "^1.4.2",
-    "@vanilla-extract/private": "^1.0.3",
     "@vanilla-extract/css-utils": "0.1.2",
+    "@vanilla-extract/private": "^1.0.3",
```
Comment (DanielSinclair) [inline packages/rainbowkit/package.json:55]: Revert

File: packages/rainbowkit/package.json [hunk 5]
```diff
@@ -61,8 +61,10 @@
     "vitest": "^0.5.0"
   },
   "dependencies": {
+    "@gnosis.pm/safe-apps-wagmi": "*",
```
Comment (DanielSinclair) [inline packages/rainbowkit/package.json:64]: Adopt Wagmi `SafeConnector` instead or discuss differences

**Review by DanielSinclair — CHANGES_REQUESTED** [review]
Please add to `packages/example` for test dApp deployment

**Comment by benjileibo** [comment]
Wooooooo!!!

**Comment by stas** [comment]
@DanielSinclair we'd have to bump the wagmi to v0.11, are you guys ok with that? 

The PR should be using the wagmi connectors now btw. I have limited time, so if it's a larger project to update everything to latest wagmi it might take longer to wrap up the PR.

**Comment by DanielSinclair** [comment]
@stas There are some TypeScript changes, so you can rebase to #1000. That should be merged later today or tomorrow anyway.

**Comment by benjileibo** [comment]
what happened to this PR @DanielSinclair

**Comment by DanielSinclair** [comment]
@benjileibo No ETA. Might require larger changes. Working on my own experimental PR at #1028.

**Comment by DanielSinclair** [comment]
Closing in favor of cherrypicks and additional changes in #1028. The behavior seems to be working as expected but requires additional testing. Feel free to try it out.

**Pull request closed without merging**